x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/gsub.tsx

```diff
@@ -6,89 +6,75 @@
 
 import React, { FunctionComponent } from 'react';
 import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCode } from '@elastic/eui';
 
-import {
-  FieldConfig,
-  FIELD_TYPES,
-  fieldValidators,
-  ToggleField,
-  UseField,
-  Field,
-} from '../../../../../../shared_imports';
-import { TextEditor } from '../field_components';
-
-const { emptyField } = fieldValidators;
+import { FIELD_TYPES, fieldValidators, UseField, Field } from '../../../../../../shared_imports';
 
-const fieldConfig: FieldConfig = {
-  type: FIELD_TYPES.TEXT,
-  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.fieldFieldLabel', {
-    defaultMessage: 'Field',
-  }),
-  validations: [
-    {
-      validator: emptyField(
-        i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.fieldRequiredError', {
-          defaultMessage: 'A field value is required.',
-        })
-      ),
-    },
-  ],
-};
+import { TextEditor } from '../field_components';
 
-const patternConfig: FieldConfig = {
-  type: FIELD_TYPES.TEXT,
-  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternFieldLabel', {
-    defaultMessage: 'Pattern',
-  }),
-  validations: [
-    {
-      validator: emptyField(
-        i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternRequiredError', {
-          defaultMessage: 'A pattern value is required.',
-        })
-      ),
-    },
-  ],
-};
+import { FieldsConfig } from './shared';
+import { FieldNameField } from './common_fields/field_name_field';
+import { IgnoreMissingField } from './common_fields/ignore_missing_field';
+import { TargetField } from './common_fields/target_field';
 
-const replacementConfig: FieldConfig = {
-  type: FIELD_TYPES.TEXT,
-  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.replacementFieldLabel', {
-    defaultMessage: 'Replacement',
-  }),
-  validations: [
-    {
-      validator: emptyField(
-        i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.replacementRequiredError', {
-          defaultMessage: 'A replacement value is required.',
-        })
-      ),
-    },
-  ],
-};
+const { emptyField } = fieldValidators;
 
-const targetConfig: FieldConfig = {
-  type: FIELD_TYPES.TEXT,
-  label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.targetFieldLabel', {
-    defaultMessage: 'Target field (optional)',
-  }),
-};
+const fieldsConfig: FieldsConfig = {
+  /* Required fields config */
+  pattern: {
+    type: FIELD_TYPES.TEXT,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternFieldLabel', {
+      defaultMessage: 'Pattern',
+    }),
+    deserializer: String,
+    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.gsubForm.patternFieldHelpText', {
+      defaultMessage: 'Regular expression used to match text.',
```

Comment:
Period isn't needed.
```suggestion
      defaultMessage: 'Regular expression used to match substrings in the field',
```